src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,185 @@
+package duke;
+
+import java.time.LocalDate;
+
+import duke.exceptions.DukeException;
+import duke.exceptions.InvalidCommandException;
+import duke.exceptions.InvalidTaskNumberException;
+import duke.exceptions.MissingDeadlineException;
+import duke.exceptions.MissingEventTimeException;
+import duke.exceptions.MissingTaskNameException;
+import duke.exceptions.MissingTaskNumberException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.ToDo;
+
+/**
+ * Makes sense of the user command
+ * Bridges communication between the user string and the tasks
+ */
+public class Parser {
```

Comment:
I realised that you made use of a lot of static methods. Why not try instantiating `Ui` objects?

Reply:
Hmm yes, will change that too 👍